internal/fields/validate.go

```diff
@@ -303,34 +303,36 @@ func FindElementDefinition(searchedKey string, fieldDefinitions []FieldDefinitio
 }
 
 func compareKeys(key string, def FieldDefinition, searchedKey string) bool {
-	if key == searchedKey {
-		return true
-	}
-
-	keyParts := strings.Split(key, ".")
-	searchedParts := strings.Split(searchedKey, ".")
-	if len(searchedParts) < len(keyParts) {
-		return false
-	}
-	for i, part := range keyParts {
-		if part == "*" {
+	var i int
+	var j int
+	for i = 0; i < len(key); i++ {
+		if j >= len(searchedKey) {
+			return false
+		}
+		if key[i] == searchedKey[j] {
```

Comment:
You could potentially save some bounds checks and syntax by making the outer loop `for i, k := range []byte(key) {` and using `k` in place of `key[i]` (note that the `[]byte` conversion does not allocate in this context see https://godbolt.org/z/Mhv7Kv9fq).